modules/tinymce/src/core/main/ts/newline/InsertNewLine.ts

```diff
@@ -9,43 +10,59 @@ import * as InsertBr from './InsertBr';
 import * as NewLineAction from './NewLineAction';
 
 const insert = (editor: Editor, evt?: EditorEvent<KeyboardEvent>) => {
-  NewLineAction.getAction(editor, evt).fold(
-    () => {
-      if (!editor.selection.isCollapsed()) {
-        execDeleteCommand(editor);
-      }
-      if (Type.isNonNullable(evt)) {
-        const event = fireFakeBeforeInputEvent(editor, 'insertLineBreak');
-        if (event.isDefaultPrevented()) {
-          return;
-        }
+  const lineBreak = () => {
```

Comment:
There's a lot of duplication between these 2 functions... can we make a generic based function instead? For example:

```ts
  const execute = (type: 'insertLineBreak' | 'insertParagraph') => {
    if (!editor.selection.isCollapsed()) {
      execDeleteCommand(editor);
    }
    if (Type.isNonNullable(evt)) {
      const event = fireFakeBeforeInputEvent(editor, type);
      if (event.isDefaultPrevented()) {
        return;
      }
    }

    if (type === 'insertLineBreak') {
      InsertBr.insert(editor, evt);
    } else {
      InsertBlock.insert(editor, evt);
    }

    if (Type.isNonNullable(evt)) {
      fireFakeInputEvent(editor, type);
    }
  }
```

Reply:
This is literally just refactored from what we had before into two variables. But yes I should probably improve it at the same time.